python/lsst/ctrl/bps/wms/htcondor/htcondor_service.py

```diff
@@ -124,6 +202,60 @@ def report(self, wms_workflow_id=None, user=None, hist=0, pass_thru=None):
 
         return list(run_reports.values()), message
 
+    def cancel(self, wms_id=None, pass_thru=None):
+        """Cancel single submitted workflow.
+
+        Parameters
+        ----------
+        wms_id : `str`
+            Cancel workflow matching WMS workflow id/path.
+        pass_thru : `str`
+            Constraints to pass through to HTCondor.
+
+        Returns
+        --------
+        deleted : `bool`
+            Whether successful deletion or not.  Currently, if any doubt or any
+            individual jobs not deleted, return False.
+        message : `str`
+            Any message from WMS (e.g., error details).
+        """
+        _LOG.debug("Canceling wms_id = %s", wms_id)
+
+        cluster_id = _wms_id_to_cluster(wms_id)
+        if cluster_id == 0:
+            deleted = False
+            message = "Invalid id"
+        else:
+            _LOG.debug("Canceling cluster_id = %s", cluster_id)
+            schedd = htcondor.Schedd()
+            constraint = f"ClusterId == {cluster_id}"
+            if "-forcex" in pass_thru:
```

Comment:
If `pass_thru` is set to its default value (`None`), the code will raise an exception. You need either additional check or use a different default value.